contracts/dao/USDTLpListaDistributor.sol

```diff
@@ -0,0 +1,316 @@
+  uint256 public stakePeriodFinish;
+  // stake token last update operation timestamp
+  uint256 public stakeLastUpdate;
+  // stake token reward of per token
+  uint256 public stakeRewardIntegral;
+  // stake token reward of per second
+  uint256 public stakeRewardRate;
+  // stake token reward integral for each account on last update time
+  // account -> reward integral
+  mapping(address => uint256) public stakeRewardIntegralFor;
+  // stake token pending reward for each account
+  // account -> pending reward
+  mapping(address => uint256) private stakeStoredPendingReward;
+
+  // lisUSD/USDT PancakeStableSwapTwoPool contract address
+  address constant stableSwapPool = 0xb1Da7D2C257c5700612BdE35C8d7187dc80d79f1;
+
+  // PancakeStableSwapTwoPoolInfo contract address
+  address constant poolInfo = 0x150c8AbEB487137acCC541925408e73b92F39A50;
+
+  event StakeRewardClaimed(address indexed receiver, uint256 amount);
+
+  /// @custom:oz-upgrades-unsafe-allow constructor
+  constructor() {
+    _disableInitializers();
+  }
+
+  /**
+   * @dev initialize contract
+   * @param _admin admin address
+   * @param _manager manager address
+   */
+  function initialize(
+    address _admin,
+    address _manager,
+    address _vault,
+    address _pancakeStaking,
+    address _stakeVault
+  ) external initializer {
+    require(_admin != address(0), "admin is the zero address");
+    require(_manager != address(0), "manager is the zero address");
+    require(_vault != address(0), "vault is the zero address");
+    __AccessControl_init();
+    __Pausable_init();
+
+    _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+    _setupRole(MANAGER, _manager);
+    _setupRole(VAULT, _vault);
+
+    lisUSD = IERC20(IStableSwap(stableSwapPool).coins(0));
+    usdt = IERC20(IStableSwap(stableSwapPool).coins(1));
+    lpToken = IStableSwap(stableSwapPool).token();
+    vault = IVault(_vault);
+    staking = _pancakeStaking;
+    stakeVault = _stakeVault;
+    name = string.concat("Lista-USDT-Staking", IERC20Metadata(lpToken).name());
+    symbol = string.concat("Lista USDT Staking ", IERC20Metadata(lpToken).symbol(), " Distributor");
+  }
+
+  modifier onlyStakeVault() {
+    require(msg.sender == stakeVault, "only stake vault can call this function");
+    _;
+  }
+
+  /**
+   * @dev deposit USDT
+   * @param usdtAmount amount of USDT to stake
+   */
+  function deposit(uint256 usdtAmount) external {
+    require(usdtAmount > 0, "Invalid usdt amount");
+    uint256 minLpAmount = getLpAmount(usdtAmount);
+
+    // transfer USDT to this contract
+    usdt.safeIncreaseAllowance(stableSwapPool, usdtAmount);
+    usdt.safeTransferFrom(msg.sender, address(this), usdtAmount);
+
+    uint256 actualLpAmount = IERC20(lpToken).balanceOf(address(this));
+    IStableSwap(stableSwapPool).add_liquidity([0, usdtAmount], minLpAmount);
+    actualLpAmount = IERC20(lpToken).balanceOf(address(this)) - actualLpAmount;
+
+    require(actualLpAmount > 0, "Invalid lp amount");
```

Comment:
actualLpAmount should be great than minLpAmount

Reply:
thanks. done in 7d83cb205a5f0fd0e81816e9dd3fc3d1e82d2f90